pkg/controller/baremetalhost/baremetalhost_controller_test.go

```diff
@@ -618,6 +620,166 @@ func TestProvision(t *testing.T) {
 	)
 }
 
+func TestProvisionWithHostConfig(t *testing.T) {
+	testBMCSecret := newSecret(defaultSecretName,
+		map[string]string{
+			"username": "User",
+			"password": "Pass",
+		},
+	)
+
+	testCases := []struct {
+		Scenario          string
+		Host              *metal3v1alpha1.BareMetalHost
+		UserDataSecret    *corev1.Secret
+		NetworkDataSecret *corev1.Secret
+		AssertF           func(*testing.T, *hostConfigData)
+	}{
+		{
+			Scenario: "host with user data only",
+			Host: newHost("host-user-data",
+				&metal3v1alpha1.BareMetalHostSpec{
+					BMC: metal3v1alpha1.BMCDetails{
+						Address:         "ipmi://192.168.122.1:6233",
+						CredentialsName: defaultSecretName,
+					},
+					UserData: &corev1.SecretReference{
+						Name:      "user-data",
+						Namespace: namespace,
+					},
+				}),
+			UserDataSecret: newSecret("user-data", map[string]string{"userData": "somedata"}),
+			AssertF: func(t *testing.T, hcd *hostConfigData) {
```

Comment:
Shouldn't this be testing to ensure the data values returned from UserData() and NetworkData() are correct?

Reply:
unit testing improved